src/formatDate.js

```diff
@@ -8,7 +8,47 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const split = fromFormat[3];
+  const dayOfYear = date.split(split);
+  const tab = [];
+
+  for (let i = 0; i < 3; i++) {
+    for (let j = 0; j < 3; j++) {
+      if (fromFormat[i] === toFormat[j]) {
+        tab[j] = dayOfYear[i];
+      }
+    }
+  }
+
+  let year;
+
+  for (let i = 0; i < 3; i++) {
+    if (fromFormat[i] === 'YYYY' || fromFormat[i] === 'YY') {
+      year = dayOfYear[i];
+    }
+  }
+
+  if (year.length === 2 && year < 30) {
+    year = '20' + year;
+  }
+
+  if (year.length === 2 && year >= 30) {
+    year = '19' + year;
+  }
```

Comment:
The logic for adjusting the year is separated from the loop that assigns the year, which can lead to incorrect behavior if the 'YY' format is not the last element in the 'fromFormat' array. It's better to integrate this logic directly into the loop where you handle the year conversion.